### ISSUE TITLE
$scope.$watch() doesn't fire when bound to checklist-models

### ISSUE THREAD

**Issue by greenie2600** [vitalets/checklist-model#89]
$scope.$watch() does not seem to fire when attached to checklist-models. Oddly, changes *are* reflected in DOM elements that are bound to the model.

[Here's a CodePen demonstrating this](http://codepen.io/anon/pen/bVbJMb). The same HTML is reproduced below.

    <!DOCTYPE html>

    <html ng-app="Demo">

    <head>
        <meta charset="utf-8">
        <title>checklist-model: Demonstration of $watch bug</title>
    </head>

    <body ng-controller="MainController">

        <h1>checklist-model Bug Demo</h1>

        <hr>

        <p>$scope.$watch() works on normal checkboxes. For example, there's a
        $watch function attached to this checkbox, which logs a message to the
        console whenever it runs. Try it:</p>

        <div class="checkbox">
            <label>
                <input type="checkbox" ng-model="normalCheckbox">
                Open the console and click me!
            </label>
        </div>

        <p>The current value of $scope.normalCheckbox is: {{ normalCheckbox }}</p>

        <hr>

        <p>However, $watch expressions attached to checklist-models only fire
        once, when the controller is instantiated. They do <em>not</em> fire when
        the model changes. Click on the checkboxes below, and observe that no
        messages appear in the console:</p>

        <div ng-repeat="flavor in flavors">
            <label>
                <input type="checkbox" checklist-model="selectedFlavors" checklist-value="flavor">
                {{ flavor }}
            </label>
        </div>

        <p>The current value of $scope.selectedFlavors is: {{ selectedFlavors }}</p>

        <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.2/angular.min.js"></script>
        <script src="http://vitalets.github.io/checklist-model/checklist-model.js"></script>
        <script>

            var app = angular.module( 'Demo', [ 'checklist-model' ] );

            app.controller( 'MainController', function( $scope ) {

                /**
                 * Normal checkbox
                 */

                $scope.normalCheckbox = false;

                $scope.$watch( 'normalCheckbox', function( oldValue, newValue ) {
                    console.log( '$scope.normalCheckbox changed!' );
                } );

                /**
                 * Checklist checkboxes
                 */

                $scope.flavors = [ 'chocolate', 'vanilla', 'strawberry' ];
                $scope.selectedFlavors = [];

                // This doesn't work!
                $scope.$watch( 'selectedFlavors', function( oldValue, newValue ) {
                    console.log( '$scope.selectedFlavors changed!' );
                } );

            } );

        </script>

    </body>

    </html>

**Comment by beradrian** [comment]
First of all - excellent bug report. Things would be so much easier if everyone will also create examples like the one you did - thanks.
The issue is that you're using `$watch` instead of `$watchCollection`. It works as expected, as the variable doesn't change, only its content. Try ` $scope.$watchCollection( 'selectedFlavors'` instead.

**Comment by immersiontravis** [comment]
D'oh! I will try that. Thanks for setting me straight.

**Comment by greenie2600** [comment]
D'oh! I will try that. Thanks for setting me straight.

**Comment by cyrilchapon** [comment]
Wow nice :)
